profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry_mock/tmgmt_poetry_mock.install

```diff
@@ -43,8 +43,46 @@ function tmgmt_poetry_mock_install() {
  * Implements hook_uninstall().
  */
 function tmgmt_poetry_mock_uninstall() {
-  $translator = tmgmt_translator_load('tmgmt_poetry_test_translator');
-  if (isset($translator)) {
-    entity_delete('tmgmt_translator', $translator->tid);
+  // Delete the tmgmt_poetry_test_translator.
+  // Note tmgmt_translator_load is not stable in uninstall cases; it returns
+  // FALSE while the translator exists in the DB.
+  // That is why we pass by EntityFieldQuery.
+  $query = new EntityFieldQuery();
+  $result = $query->entityCondition('entity_type', 'tmgmt_translator')
+    ->propertyCondition('name', 'tmgmt_poetry_test_translator')
+    ->execute();
+  if (!empty($result['tmgmt_translator'])) {
+    $controller = entity_get_controller('tmgmt_translator');
+    // Since the entity controller handles the deletion of the attached
+    // entities (messages, job items) we just need to invoke it directly.
+    $controller->delete(array_keys($result['tmgmt_translator']));
   }
+
+  // Clean translation jobs related to the tmgmt_poetry_test_translator.
```

Comment:
Instead of 'Clean' I propose to use 'Delete' as that would follow the name of the used function in the block code below. Also it follows the comment which is used above.